File: mini-util/src/main/java/com/mini/core/util/tree/AbstractTree.java

package com.mini.core.util.tree;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public abstract class AbstractTree<T extends ITree<T>> implements ITree<T> {

    @Override
    public final void addChild(@NotNull T child) {
        ITree.super.addChild(child);
    }

    @Override
    public void setChildren(List<T> children) {
        this.children = children;
    }

    @Override
    public List<T> getChildren() {
        return children;
    }

    private List<T> children;
}

File: mini-util/src/main/kotlin/com/mini/core/util/tree/AbstractTree.kt
@file:Suppress("unused")

package com.mini.core.util.tree

abstract class AbstractTree<T : ITree<T>> : ITree<T> {
    final override val children: MutableList<T> = mutableListOf()

    final override fun addChild(child: T) {
        super.addChild(child)
    }
}